### ISSUE TITLE
Unable to remove existing columns

### ISSUE THREAD

**Issue by JasonTheAdams** [moderntribe/advanced-post-manager#10]
Greetings!

I installed this since its required for filters now with Events Calendar Pro. Unfortunately, now that its enabled the columns I was previously hiding are no longer being hidden.

```
add_filter('manage_' . EVENT_TYPE . '_posts_columns', function($columns) {
  unset($columns['wpseo-focuskw']);
  unset($columns['wpseo-metadesc']);
  unset($columns['wpseo-score']);
  unset($columns['wpseo-title']);
  unset($columns['events-cats']);
  unset($columns['author']);
  return $columns;
}, 9999);
```

I've dug through your code quite a bit, and I'm really baffled. This is being reflected in `Tribe_Columns->get_column_headers`, but when it finally renders, there they are.

**Comment by borkweb** [comment]
My guess is that `EVENT_TYPE` isn't defined anywhere. When I change that snippet to the following, it seems to work fine!

```php
add_filter('manage_' . Tribe__Events__Main::POSTTYPE . '_posts_columns', function( $columns ) {
  unset($columns['wpseo-focuskw']);
  unset($columns['wpseo-metadesc']);
  unset($columns['wpseo-score']);
  unset($columns['wpseo-title']);
  unset($columns['events-cats']);
  unset($columns['author']);
  return $columns;
}, 9999);
```

**Comment by JasonTheAdams** [comment]
Hi @borkweb, thanks for replying!

Sorry, I have EVENT_TYPE defined in the functions.php. I'm confident that's not the issue, but I tried your line as well as avoiding constants altogether and using tribe_events. No luck.

Out of curiosity, did you have Wordpress SEO enabled where you tested it? Those are primarily where I'm having the issue. I unset the header, and nothing happens. *But* if I disable this plugin, they go away. :confused: 

Strangely enough, if I go to [Tribe_Columns::get_column_headers (line 377)](https://github.com/moderntribe/advanced-post-manager/blob/release/119/lib/tribe-columns.class.php#L377) and return an empty array instead of `return $headers`, every header (not column entirely) goes blank *except* the SEO headers... which kind of breaks my brain -- where the heck are the headers coming from!? Hahah!

**Comment by borkweb** [comment]
Ohhhh! Gotcha. I'm thinking that a clean approach would be to unhook the SEO filter that adds those columns. Try something like this:

```php
add_action( 'current_screen', function( $screen ) {
	if ( ! isset( $GLOBALS['wpseo_metabox'] ) || 'edit-tribe_events' !== $screen->id ) {
		return;
	}

	remove_filter( 'manage_' . Tribe__Events__Main::POSTTYPE . '_posts_columns', array( $GLOBALS['wpseo_metabox'], 'column_heading' ), 10, 1 );
} );
```

That hook is being created [here](https://github.com/Yoast/wordpress-seo/blob/trunk/admin/class-metabox.php#L130)

**Comment by JasonTheAdams** [comment]
Thanks for the snippet, that worked! Admittedly, I'm a bit concerned that this plugin is written in such a way that core functionality, like this, becomes broken. It seems to hijack the table view in its entirety, and I'm not sure how I feel about that. I do like what you're doing, I'm just not thrilled about how you're getting there.

**Comment by borkweb** [comment]
Hey! Glad that worked :+1:

I've been looking into this issue a bit more and your point about broken core functionality is spot on. Ideally, your original snippet _should_ work. I've tracked down the bug and will be submitting a Pull Request shortly!  As to the point about hijacking the table - I'm not sure I know what you mean there. Advanced Post Manager's whole purpose is to inject fields into the default list table, which it seems to be doing happily (especially once this bug you've found is fixed!).

**Comment by zbtirrell** [comment]
Matt made an attempt at this fix last summer with this pull request: https://github.com/moderntribe/advanced-post-manager/pull/14 but we discovered larger architectural restrictions. We have logged this in our internal ticket tracking system so that it is not lost. However, we don't monitor GitHub Issues and had forgotten to disable it here.

In the future, we'd be happy to take bug reports in the WordPress.org support forums - https://wordpress.org/support/plugin/advanced-post-manager
